frame/support/procedural/src/pallet/parse/pallet_struct.rs

```diff
@@ -84,7 +84,10 @@ impl syn::parse::Parse for PalletStructAttr {
 			syn::parenthesized!(generate_content in content);
 			let vis = generate_content.parse::<syn::Visibility>()?;
 			generate_content.parse::<syn::Token![trait]>()?;
-			let keyword = generate_content.parse::<keyword::Store>()?;
+			let mut keyword = generate_content.parse::<keyword::Store>()?;
+			// set span of Store keyword same to attirbute macro `generate_store`
+			// so that deprecated warning highlights use of macro.
+			keyword.span = content.span();
 			Ok(Self::GenerateStore { vis, keyword, span })
```

Comment:
If anything, this `span` variable should be updated to the span of the entire attribute, and not by mutating the span of the `Store` keyword`.

In general, it's best not to modify the meaning of what the fields represent. A span in the keyword means a span for the keyword; modifying it to represent anything else to solve a problem is considered a hack.

Reply:
https://github.com/paritytech/substrate/pull/13535/commits/ed2a1707d532298ea1e4292b6313d5e2506d2154